packages/react/src/components/TextField/TextField.types.ts

```diff
@@ -278,6 +278,14 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
    */
   canRevealPassword?: boolean;
 
+  /**
+   * Whether to use the default password reveal button. This only applies to Edge and IE and using the default
+   * reveal icon will show both IE/Edge's button and Fluent UI's button.
+   *
+   * @defaultvalue false
+   */
+  enableBuiltInRevealButton?: boolean;
+
   /**
    * If `canRevealPassword` is true, aria label for the reveal password button (example: "Show
    * password"). Note that this will NOT be used in browsers known to have a built-in reveal
```

Comment:
This comment lead me to dig deeper into this issue and [this PR](https://github.com/microsoft/fluentui/commit/3a9fd36f3096ca19c100c956dcddb4196cf9ef08#diff-abaa20fc65c181b1cd4b9ae6857c16ff4cd6f7ee6f2a8d7ed96c50e9291610ba) seems to have broken the behavior.

Fundamentally the bug is that [we're sniffing the user agent](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/TextField/TextField.base.tsx#L675) rather than feature detecting but changing the regex will "fix" the behavior and only show a single reveal button (change `^Edg` to `Edg`).

Off the top of my head I see two things we can do here:

1. Change this PR to fix the bug and remove the new prop, essentially restoring the old behavior.
2. Keep the new prop and behavior from this PR and remove the old UA sniffing.

Reply:
I should add that I'm not sure it's possible to feature detect the password reveal button so UA sniffing may be the best we can do.